src/inmanta/agent/forking_executor.py

```diff
@@ -435,9 +440,28 @@ async def execute(
         reason: str,
     ) -> None:
         await self.connection.call(ExecuteCommand(gid, resource_details, reason))
+        self.check_modification_time_venv()
+
+    def check_modification_time_venv(self) -> None:
+        """
+        Make sure that the time of most recent content modification of the `inmanta_env_status` file is within the last two
+        hours. If it's not, the file is overwritten.
+        """
+        inmanta_env_status = os.path.join(self.executor_virtual_env.env_path, ".inmanta_env_status")
```

Comment:
1. I think it is simpler to just update it always  (e.g. using https://docs.python.org/3/library/pathlib.html#pathlib.Path.touch)  (it makes it more accurate and requires less calls to the OS)

2. I think this is risky: if an executor is running in the venv, but not doing anything, the venv can still be removed. I think it is easier to just start a timer (Inside the executor, not on the agent side like you did now) that updates the file every minute (I propose one minute, because it is sufficiently fast for all cases and sufficiently slow to not matter at all wrt performance)

Reply:
Going back to the second point: but now we could have a race condition between the touch of the file and the removal of the old virtual envs right?

Reply:
As it is now, we can only race when the executor is starting.